sdk/metric/internal/aggregate/sum.go

```diff
@@ -269,24 +214,19 @@ func (s *precomputedDeltaSum[N]) Aggregation() metricdata.Aggregation {
 		DataPoints:  make([]metricdata.DataPoint[N], 0, len(s.values)),
 	}
 	for attr, value := range s.values {
-		v := value.measured + value.filtered
-		delta := v - s.reported[attr]
+		delta := value - s.reported[attr]
 		out.DataPoints = append(out.DataPoints, metricdata.DataPoint[N]{
 			Attributes: attr,
 			StartTime:  s.start,
 			Time:       t,
 			Value:      delta,
 		})
-		if delta != 0 {
-			s.reported[attr] = v
-		}
-		value.filtered = N(0)
-		s.values[attr] = value
-		// TODO (#3006): This will use an unbounded amount of memory if there
-		// are unbounded number of attribute sets being aggregated. Attribute
-		// sets that become "stale" need to be forgotten so this will not
-		// overload the system.
+		newReported[attr] = value
+		// Unused attribute sets do not report.
+		delete(s.values, attr)
 	}
+	// Unused attribute sets are forgotten.
+	s.reported = newReported
```

Comment:
I would like feedback on this change in particular.  It only keeps track of the previous collection interval's values for the purpose of computing deltas in subsequent collection intervals. 
This has the benefit of ensuring that "churning" attribute sets don't cause a memory leak in precomputed delta sums.  Without the change, we would continue to track the previous value for an attribute set forever.